From 49c764606a943724b327064a34d1333363bebd1f Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Mon, 18 Aug 2014 19:55:53 -0400
Subject: [scsi] qla2xxx: Print proper QLAFX00 product name at probe

Message-id: <1408391817-6389-6-git-send-email-cdupuis@redhat.com>
Patchwork-id: 87904
O-Subject: [PATCH RHEL 7.1 e-stor 05/69] qla2xxx: Print proper QLAFX00 product name at probe.
Bugzilla: 1089346
RH-Acked-by: Rob Evers <revers@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Armen Baloyan <armen.baloyan@qlogic.com>

scsi commit id a324031c4373fd9df55b771f8253f5df44371680

Signed-off-by: Armen Baloyan <armen.baloyan@qlogic.com>
Signed-off-by: Saurav Kashyap <saurav.kashyap@qlogic.com>
Signed-off-by: James Bottomley <JBottomley@Parallels.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/qla2xxx/qla_dbg.c b/drivers/scsi/qla2xxx/qla_dbg.c
index ee5c183..acc1ea4 100644
--- a/drivers/scsi/qla2xxx/qla_dbg.c
+++ b/drivers/scsi/qla2xxx/qla_dbg.c
@@ -11,7 +11,7 @@
  * ----------------------------------------------------------------------
  * |             Level            |   Last Value Used  |     Holes |
  * ----------------------------------------------------------------------
- * | Module Init and Probe        |       0x0159       | 0x4b,0xba,0xfa |
+ * | Module Init and Probe        |       0x015a       | 0x4b,0xba,0xfa |
  * | Mailbox commands             |       0x1181       | 0x111a-0x111b  |
  * |                              |                    | 0x1155-0x1158  |
  * |                              |                    | 0x1018-0x1019  |
diff --git a/drivers/scsi/qla2xxx/qla_os.c b/drivers/scsi/qla2xxx/qla_os.c
index 24f5265..4c225db 100644
--- a/drivers/scsi/qla2xxx/qla_os.c
+++ b/drivers/scsi/qla2xxx/qla_os.c
@@ -2850,8 +2850,12 @@ skip_dpc:
 
  qla2x00_dfs_setup(base_vha);
 
- ql_log(ql_log_info, base_vha, 0x00fb,
-     "QLogic %s - %s.\n", ha->model_number, ha->model_desc);
+ if (IS_QLAFX00(ha))
+  ql_log(ql_log_info, base_vha, 0x015a,
+      "QLogic %s.\n", ha->mr.product_name);
+ else
+  ql_log(ql_log_info, base_vha, 0x00fb,
+      "QLogic %s - %s.\n", ha->model_number, ha->model_desc);
  ql_log(ql_log_info, base_vha, 0x00fc,
      "ISP%04X: %s @ %s hdma%c host#=%ld fw=%s.\n",
      pdev->device, ha->isp_ops->pci_info_str(base_vha, pci_info),
-- 
1.7.1